http-client/src/main/java/io/micronaut/http/client/MatchingFilterResolver.java

```diff
@@ -0,0 +1,171 @@
+    private final AnnotationMetadataResolver annotationMetadataResolver;
+    private final Set<String> clientIdentifiers;
+
+    /**
+     * @param annotationMetadata         annotation metadata of a declarative http client
+     * @param clientIdentifiers          client identifiers associated with a client
+     * @param annotationMetadataResolver annotation medata resolver
+     * @param filters                    all filters
+     */
+    @Inject
+    MatchingFilterResolver(@Nullable @Parameter AnnotationMetadata annotationMetadata,
+                           @Nullable @Parameter Set<String> clientIdentifiers,
+                           @Nullable AnnotationMetadataResolver annotationMetadataResolver,
+                           Collection<HttpClientFilter> filters) {
+        this.filters = filters;
+        this.annotationMetadata = Optional.ofNullable(annotationMetadata).orElse(AnnotationMetadata.EMPTY_METADATA);
+        this.annotationMetadataResolver = Optional.ofNullable(annotationMetadataResolver)
+                .orElse(AnnotationMetadataResolver.DEFAULT);
+        this.clientIdentifiers = Optional.ofNullable(clientIdentifiers).orElse(Collections.emptySet());
+    }
+
+    /**
+     * @param request    http request
+     * @param requestURI path URI
+     * @return filters that should be applied to request
+     */
+    public List<HttpClientFilter> resolveFilters(io.micronaut.http.HttpRequest<?> request,
+                                          URI requestURI) {
+        List<HttpClientFilter> filterList = new ArrayList<>();
+        String requestPath = StringUtils.prependUri("/", requestURI.getPath());
+        io.micronaut.http.HttpMethod method = request.getMethod();
+        for (HttpClientFilter filter : filters) {
+            if (filter instanceof Toggleable && !((Toggleable) filter).isEnabled()) {
+                continue;
+            }
+
+            Class<? extends Annotation>[] clientDeclaredFilterAnnotations = (Class<? extends Annotation>[]) annotationMetadata
+                    .getAnnotationTypesByStereotype(FilterAnnotation.class)
+                    .stream()
+                    .toArray(Class[]::new);
+
+            Class<? extends Annotation>[] declaredFilterAnnotations = (Class<? extends Annotation>[]) annotationMetadataResolver.resolveMetadata(filter)
+                    .getAnnotationTypesByStereotype(FilterAnnotation.class)
+                    .stream()
+                    .toArray(Class[]::new);
+
+            Optional<AnnotationValue<Filter>> filterAnnotationMetadataOpt = annotationMetadataResolver.resolveMetadata(filter)
+                    .findAnnotation(Filter.class);
+
+            FilterProperties annotationFilterProperties = filterAnnotationMetadataOpt
+                    .map(f -> new FilterProperties(
+                            f.stringValues("patterns"),
+                            f.get("methods", io.micronaut.http.HttpMethod[].class, null),
```

Comment:
Please use `enumValues("methods", HttpMethod[].class)`

Reply:
@graemerocher already answered on this comment earlier, but rechecked methods in AnnotationValue and there is no such method that returns array/collection of Enums. There are only methods for single enum